fix/workorder-detail-fix.lua

```diff
@@ -0,0 +1,199 @@
+                job_item, order_item.item_type, order_item.item_subtype )
+            if suitable then
+                correct_item_details(job_item, order_item)
+                modified = true
+            else --[[ error on unsuitable item?]] end
+        end
+    end
+    if modified then jobs_corrected = jobs_corrected + 1 end
+end
+
+-- check orders list to see if fix is needed
+-- yields id-order table for the orders with issues, or nil when no issues
+local PrepareMeal = df.job_type.PrepareMeal
+local SewImage = df.job_type.SewImage
+local NONE = df.job_type.NONE
+local function get_bugged_orders()
+    local orders = {}
+    local num_bugged, improve_item_index, item = 0, 1, nil
+    for _, order in ipairs(df.global.world.manager_orders) do
+        if not order.items then goto nextorder end
+        local order_job_type = order.job_type
+        if not offending_jobs[order_job_type] then goto nextorder end
+        if #order.items == 0 then goto nextorder end -- doesn't happen
+
+        -- for PrepareMeal jobs, any one of the items could be an issue.
+        if order_job_type == PrepareMeal then
+            for _, _item in ipairs(order.items) do
+                if _item.item_type ~= NONE then goto fix end
+            end
+            goto nextorder
+        end
+
+        -- All other types are improve jobs; only the improved item is checked
+        -- Only SewImage has the item-to-improve at items[0]
+        improve_item_index = (order_job_type ~= SewImage) and 1 or 0
+        item = order.items[improve_item_index]
+        if not item then goto nextorder end -- error here?
+        if item.item_type == NONE then goto nextorder end
+        :: fix ::
+        num_bugged = num_bugged + 1
+        orders[order.id] = order
+        :: nextorder ::
+    end
+    if num_bugged == 0 then return nil end
+    return orders
+end
+
+-- correct newly dispatched work order jobs
+local disable
+local function on_dispatch_tick()
+    if df.global.cur_year_tick % 150 ~= 30 then
+        print(script_name.." desynced from dispatch tick")
+        repeatutil.cancel(schedule_key)
+        disable(true)
+    end
+    if #managers == 0 then return end
+    if df.global.plotinfo.manager_timer ~= 10 then return end
+    local orders = get_bugged_orders()
+    if not orders then return end -- no bugs to fix
+    local highest = last_job_id
+    for _, job in utils.listpairs(df.global.world.jobs.list) do
+        if job.id <= last_job_id then goto nextjob end
+        highest = math.max(job.id, highest)
+        if job.order_id == -1 then goto nextjob end
+        local order = orders[job.order_id]
+        if not order then goto nextjob end -- order wasn't bugged
+        -- job in progress: only happens on the first run.
+        -- experimental fix: remove the items and un-task them
+        while #job.items ~= 0 do
+            job.items[0].item.flags.in_job = false
+            job.items:erase(0)
+        end
```

Comment:
needs to remove specific refs on the item; see my previous comment on exposing `Job::disconnectJobItem`